presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -238,6 +238,7 @@
     private String nativeExecutionExecutablePath = "./presto_server";
     private boolean randomizeOuterJoinNullKey;
     private boolean isOptimizeConditionalAggregationEnabled;
+    private boolean isRemoveRedundantDistinctAggregationEnabled = true;
```

Comment:
are we sure we want to enable it by default?

Reply:
> are we sure we want to enable it by default?

Yes this is a very general optimization that should always help and we are being very conservative and also adding more tests for making sure it works for different patterns.

Reply:
My main worry is that if there is a bug in the code and it would big pain to do fixes in prod. If we have high confidence with full correctness in verifiers, I'm also ok either way.

Reply:
>  Yes - we will do some targeted verifier runs as the pattern is relatively easy to look for in the logs.

Yeah, I will run verifier test and report here.

Reply:
I get about 40 queries which trigger this optimization, with most queries showing about 20% reduction in cpu time.